src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -169,6 +176,9 @@ async def _run_impl(self) -> None:
         Raises:
             asyncio.CancelledError: if the MovingWindow task is cancelled.
         """
+        received_samples_count = 0
+        wait_for_samples_sender = self._wait_for_samples_channel.new_sender()
```

Comment:
I guess once a MW was stopped there is no way to start it again, right? Otherwise the sender should probably be created in the constructor instead to avoid *leaking* sender objects.